### PR TITLE
Remove more legacy deprecations

### PR DESCRIPTION
**Pull request by ausi**

### CODE REVIEW

File: core-bundle/src/Resources/contao/classes/DataContainer.php [hunk 1]
```diff
@@ -429,55 +429,8 @@ protected function row($strPalette=null)
...
-
-			// Deprecated since Contao 4.2, to be removed in Contao 5.0
-			if (Input::post($this->strInputName) === null && \in_array($this->strInputName, $paletteFields))
-			{
-				trigger_deprecation('contao/core-bundle', '4.2', 'Using $_POST[\'FORM_FIELDS\'] has been deprecated and will no longer work in Contao 5.0. Make sure to always submit at least an empty string in your widget.');
-			}
-
 			// Validate and save the field
-			if ($objWidget->submitInput() && (\in_array($this->strInputName, $paletteFields) || Input::get('act') == 'overrideAll'))
+			if ($objWidget->submitInput() && (Input::post($this->strInputName) !== null || Input::get('act') == 'overrideAll'))
```
Comment (ausi) [inline core-bundle/src/Resources/contao/classes/DataContainer.php]: This needs verification of someone familiar with the data container and `$_POST['FORM_FIELDS']` I think ☺️

Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/DataContainer.php]: I‘m not sure if we should remove all of the logic. I have created another PR to remove the deprecated `FORM_FIELDS` feature (see #4740), which keeps the "only submit fields that are in the palette" logic. Depending on the discussion there, we can decide how to proceed here.

Comment (ausi) [reply core-bundle/src/Resources/contao/classes/DataContainer.php]: Done.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @ausi.

**Pull request merged**